invenio_users_resources/services/groups/service.py

```diff
@@ -23,7 +23,7 @@ class GroupsService(RecordService):
     def read(self, identity, id_):
         """Retrieve a user group."""
         # resolve and require permission
-        group = GroupAggregate.get_record(id_)
+        group = GroupAggregate.get_record_by_name(id_)
```

Comment:
Comment: I am not fully convinced about overriding the `read` function and changing the "type" of the parameter `id_`, I guess it's my Java mentality ... 

Anyway I would update the error message below to f"No group with name '{id_}'."

Reply:
the user sees it as an id, that's why the message is not changed. We hid the real ids of groups in order not to expose the incremental counter in the API but this is transparent to the user